modules/apps/map/map-common/src/main/resources/META-INF/resources/js/MapBase.js

```diff
@@ -62,15 +60,39 @@ const pendingCallbacks = {};
  * @abstract
  * @review
  */
-class MapBase extends State {
+class MapBase extends EventEmitter {
+	get position() {
+		return this._STATE_.position;
+	}
+
+	set position(position) {
+		this.emit('positionChange', {
+			newVal: {
+				address: position.address,
+				location: position.location,
+			},
+		});
+
+		this._STATE_.position = position;
+	}
 
 	/**
 	 * MapBase constructor
 	 * @param  {Array} args List of arguments to be sent to State constructor
 	 * @review
 	 */
-	constructor(...args) {
-		super(...args);
+	constructor({boundingBox, controls, geolocation, position}) {
+		super({boundingBox, controls, geolocation, position});
```

Comment:
I think you would be better off doing like this


```suggestion
	constructor(args) {
		super(args);
		
		const {boundingBox, controls, geolocation, position} = args;
```

Because we don't really know exactly what the extended class requires in it's super, so this way we can pass everything through.

Reply:
Also, can we add some default values here for these since they have default values set in STATE previously?

I believe you set these in your tests, so if we set the default values here then we won't need them in the tests.

```
		boundingBox: {},
		controls: [],
		geolocation: true,
		position: {},
```

Reply:
Sound good, I'm not proficient with classes so I wasn't 100% sure how it all works under the hood.

Reply:
```js
constructor(args) {
	super(args);

	const {
		boundingBox = {},
		controls = [],
		geolocation = true,
		position = {},
		tileURI = '//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
	} = args;

        ...
```

@bryceosterhaus This still causes the issue in the test where the test isn't aware of the `boundingBox` and other args. 

`mapImpl = new MapImpl();` is how the MapBase class is being invoked in the test.

Reply:
I think you also need to default args like this `constructor(args = {})`